src/Umbraco.Web.UI.Client/src/common/resources/datatype.resource.js

```diff
@@ -30,369 +30,412 @@ function dataTypeResource($q, $http, umbDataFormatter, umbRequestHelper) {
+     *
+     */
+    save: function (dataType, preValues, isNew) {
+
+      var saveModel = umbDataFormatter.formatDataTypePostData(dataType, preValues, "save" + (isNew ? "New" : ""));
+
+      return umbRequestHelper.resourcePromise(
+        $http.post(umbRequestHelper.getApiUrl("dataTypeApiBaseUrl", "PostSave"), saveModel),
+        "Failed to save data for data type id " + dataType.id);
+    },
+
+    /**
+     * @ngdoc method
+     * @name umbraco.resources.dataTypeResource#move
+     * @methodOf umbraco.resources.dataTypeResource
+     *
+     * @description
+     * Moves a node underneath a new parentId
+     *
+     * ##usage
+     * <pre>
+     * dataTypeResource.move({ parentId: 1244, id: 123 })
+     *    .then(function() {
+     *        alert("node was moved");
+     *    }, function(err){
+     *      alert("node didnt move:" + err.data.Message); 
+     *    });
+     * </pre> 
+     * @param {Object} args arguments object
+     * @param {Int} args.idd the ID of the node to move
+     * @param {Int} args.parentId the ID of the parent node to move to
+     * @returns {Promise} resourcePromise object.
+     *
+     */
+    move: function (args) {
+      if (!args) {
+        throw "args cannot be null";
+      }
+      if (!args.parentId) {
+        throw "args.parentId cannot be null";
+      }
+      if (!args.id) {
+        throw "args.id cannot be null";
+      }
+
+      return umbRequestHelper.resourcePromise(
+        $http.post(umbRequestHelper.getApiUrl("dataTypeApiBaseUrl", "PostMove"),
+          {
+            parentId: args.parentId,
+            id: args.id
+          }, { responseType: 'text' }),
+        'Failed to move content');
+    },
+
+    /**
+     * @ngdoc method
+     * @name umbraco.resources.dataTypeResource#copy
+     * @methodOf umbraco.resources.dataTypeResource
+     *
+     * @description
+     * Copies a node underneath a new parentId
+     *
+     * ##usage
+     * <pre>
+     * dataTypeResource.copy({ parentId: 1244, id: 123 })
+     *    .then(function() {
+     *        alert("node has been copied");
+     *    }, function(err){
+     *      alert("node didnt copy:" + err.data.Message);
+     *    });
+     * </pre>
+     * @param {Object} args arguments object
+     * @param {Int} args.idd the ID of the node to copy
+     * @param {Int} args.parentId the ID of the parent node to copy to
+     * @returns {Promise} resourcePromise object.
+     *
+     */
+    copy: function (args) {
+      if (!args) {
+        throw "args cannot be null";
+      }
+      if (!args.parentId) {
+        throw "args.parentId cannot be null";
+      }
+      if (!args.id) {
+        throw "args.id cannot be null";
+      }
+
+      return umbRequestHelper.resourcePromise(
+        $http.post(umbRequestHelper.getApiUrl("dataTypeApiBaseUrl", "PostCopy"),
+          {
+            parentId: args.parentId,
+            id: args.id
+          }, { responseType: 'text' }),
+        'Failed to copy content');
```

Comment:
Could this error message be localized?

Reply:
Do you have an example of how to? I see that the rest of the file also doesn't have localized messages, so maybe this is a bit out of scope.

Reply:
Woops did not pay attention  to the other error handlers. Let's keep it consistent then and let plain English in there for now😁